File: greedy.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    float change;
    
    do
    {
        printf("enter the amount owed to you: ");
        change = get_float();
    }
    while (change <= 0);
    
    int x;
    
    x = change * 100;
    
    int quarter = 0;

    while (x >= 25)
    {
       quarter = quarter + 1;
       x = x - 25;
    }

    int dime = 0;     //number of dimes to start off
    int y = x % 25;   //remaining balance in cents
    
    
    while (y >= 10)
    {
        dime = dime + 1;   //increase number of dimes by 1  
        y = y - 10;         //decrease the total amount owed by 10 cents 
    }
    
    int nickel = 0;
    int z = y % 10;
    
    while (z >=5)
    {
        nickel = nickel +1; 
        z = z - 5;
    }
    
    int penny = 0;
    int p = z % 5;
    
    while (p >=1)
    {
        penny = penny +1; 
        p = p - 1;
    }
    
    printf("%i\n", quarter + dime + nickel + penny);
    
}

File: greedy.py
'''
Pset6 - Greedy (less comfortable)
Calculate the minumum number of coins to give user change 
CS50
'''
import cs50
import math

def main():
    # get the total amount of change required from user
    amount = get_amount_float()
    
    # convert the change required to cents
    cents = round(amount * 100)
    
    # initialize the number of coins needed
    n = 0
    
    # increase the number of coins needed by the the maximum number of quarters 
    n += cents // 25
    
    # calculate the remaining amount of change 
    cents = cents % 25
    
    n += cents // 10
    cents = cents % 10 
    
    n += cents // 5
    cents = cents % 5
    
    n += cents / 1 
    
    print(n)
    

# define the class that asks for user input 
def get_amount_float(): 
    while True:
        print("Amount owed: ", end="")
        n = cs50.get_float()
        if n >= 0:
            break
    return n

# implement main 
if __name__ == "__main__":
    main()